src/OSPSuite.Core/Comparison/WithValueOriginComparison.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using OSPSuite.Core.Domain;
+
+namespace OSPSuite.Core.Comparison
+{
+   public class WithValueOriginComparison<T> where T : class, IWithValueOrigin
+   {
+      public void AddValueOriginToComparison(IComparison<T> comparison, DiffBuilder<T> diffBuilder, Action<IComparison<T>> comparisonAction)
+      {
+         var numberOfItemsInComparison = comparison.Report.Count;
+
+         comparisonAction(comparison);
+
+         if (!shouldAddValueOriginToComparison(numberOfItemsInComparison, comparison))
+            return;
+
+         diffBuilder.CompareValues(x => x.ValueOrigin, x => x.ValueOrigin, comparison);
+      }
+
+      private bool shouldAddValueOriginToComparison(int numberOfItemsInComparisonBeforeStartingComparison, IComparison<T> comparison)
```

Comment:
Strategy that defines if ValueOrigin should be shown or not
If we compare everything, I would suggest to show the difference no matter if the show value origin flag is set or not. (otherwise there is NO way to see that difference)

Otherwise, if the comparison actually performed some changes, we return the setting value